Source/Common/MicroArgParser.h

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include <optional>
+#include <string_view>
+#include <span>
+#include "FEXCore/fextl/vector.h"
+
+namespace FEX {
+class MicroArgParser final {
+public:
+  class Value final {
+  public:
+    Value() = default;
+    constexpr void Set(std::string_view value) {
+      HasValue = true;
+      _Value.clear();
+      _Value.emplace_back(value);
+    }
+
+    constexpr void Append(std::string_view value) {
+      HasValue = true;
+      _Value.emplace_back(value);
+    }
+
+    template<typename T>
+    T Get(std::string_view Default) const;
+
+
+    const fextl::vector<std::string_view>& GetVector() const {
+      return _Value;
+    }
+
+  private:
+    fextl::vector<std::string_view> _Value;
```

Comment:
Identifiers starting with an underscore look out of place here.

Also, this can be an `std::optional<vector<string_view>>` instead to avoid having two variables to manage the same state.